### PR TITLE
Consider system image for quick roll forward

### PR DESCRIPTION
**Pull request by yuying-y**
b/310711891

Change-Id: I34bea666b2780402775f0f8793474b474d5bc06c

### CODE REVIEW

**Comment by yuying-y** [comment]
Verified on Linux.

File: components/update_client/cobalt_slot_management.cc [hunk 1]
```diff
@@ -300,7 +300,7 @@ bool CobaltQuickUpdate(
   int slot_candidate = -1;
 
   // Iterate over all writeable slots - index >= 1.
```
Comment (y4vor) [inline components/update_client/cobalt_slot_management.cc:302]: Update the comment?

File: components/update_client/cobalt_slot_management.cc [hunk 2]
```diff
@@ -329,12 +329,13 @@ bool CobaltQuickUpdate(
...
-               starboard::loader_app::AnyGoodAppKeyFile(
-                   installation_dir.value().c_str())) {
+               (i == 0 || !DrainFileIsAnotherAppDraining(
+                              installation_dir.value().c_str(), app_key) &&
+                              !CheckBadFileExists(
+                                  installation_dir.value().c_str(), app_key) &&
+                              !SbFileExists(good_app_key_file_path.c_str()) &&
+                              starboard::loader_app::AnyGoodAppKeyFile(
+                                  installation_dir.value().c_str()))) {
       // Found a slot with newer version than the current version that's not
```
Comment (y4vor) [inline components/update_client/cobalt_slot_management.cc:339]: This comment needs updating too

File: components/update_client/cobalt_slot_management.cc [hunk 3]
```diff
@@ -329,12 +329,13 @@ bool CobaltQuickUpdate(
       continue;
     } else if (slot_candidate_version < installed_version &&
```
Comment (y4vor) [inline components/update_client/cobalt_slot_management.cc:330]: Is the `slot_candidate_version` reliable?

Comment (yuying-y) [reply components/update_client/cobalt_slot_management.cc:330]: Yes, `slot_candidate_version` is a place holder of the highest version found. It starts with `1.0.1`. `installed_version` is the version read from `manifest.json` of the current iterated slot. It works for system image too. At the moment, we have to trust the `manifest.json` in the system image to be accurately reflecting the Evergreen version, as we don't have other ways to know the version before actually loading the binary.

**Review by y4vor — APPROVED** [review]

**Pull request merged**